File: two-sum/1.two-sum.java

class Solution {

    public int[] twoSum(int[] nums, int target) {

        Map<Integer, Integer> result = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int goal = target - nums[i];

            if (result.containsKey(goal)) {
                return new int[]{result.get(goal), i};
            }
            result.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: two-sum/1.two-sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}

        for i, value in enumerate(nums):
            goal = target - value

            if goal in hashmap:
                return [hashmap[goal], i]
            hashmap[value] = i
